### PR TITLE
[Sajal] iP

### PR DESCRIPTION
**Pull request by SajalVaishnav**
#ChampakChacha2.0

Chacha ji once said:
> Arre babuchat JAA NAHANE JAA

With a champak chacha behind your back, no need to

* worry about forgetting your tasks,
* or deadlines,
* or events!!
* or bathing *913287498 times* a day

All you need to get your very own champak chacha is,

1. download it from here.
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you

And it is **FREE**

Features:

- [x] maintains your tasks 
- [x] reminds you of deadlines
- [ ] ghar mein bujurg laata h:family_man_man_girl_boy:
- [x] maintains your events 

If you know java you can even set the champak chacha dp. Change the following lines in `Duke.java`:
```
 private Image user = new Image(this.getClass().getResourceAsStream("/images/DaUser.png"));
    private Image duke = new Image(this.getClass().getResourceAsStream("/images/DaDuke.png"));```

### CODE REVIEW

File: src/main/java/duke/Deadline.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    public void markAsDone() {
+        this.done = true;
+    }
+
+    public String getTaskString() {
+        if(done) {
+            return "[D][X] " + this.taskString + " (by: " + this.deadline.format(DateTimeFormatter.ofPattern("MMM dd yyyy")) + ")";
+        } else {
+            return "[D][ ] " + this.taskString + " (by: " + this.deadline.format(DateTimeFormatter.ofPattern("MMM dd yyyy")) + ")";
```
Comment (eeliana) [inline src/main/java/duke/Deadline.java]: Perhaps repeated code code can be extracted into a variable?

File: src/main/java/duke/DeleteTaskMessage.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    private int taskIndex;
+
+    public DeleteTaskMessage(int index) {
+        this.taskIndex = index;
+        this.deletedTask = TaskList.getTaskList().getTasks().remove(taskIndex);
+    }
+    public void display() {
+        System.out.println("Theek h bhai... ye task hata diya");
+        System.out.println(deletedTask.getTaskString());
+        System.out.println("Ab " + TaskList.getTaskList().getSize() + " tasks bache list mein.");
```
Comment (eeliana) [inline src/main/java/duke/DeleteTaskMessage.java]: Should these be in English?

File: src/main/java/duke/Event.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package duke;
+
+/**
+ * The class for creating and managing Events added to TaskList. Extends Task.
+ */
+public class Event extends Task{
+    private boolean done;
```
Comment (eeliana) [inline src/main/java/duke/Event.java]: Should the boolean be named as isDone instead?

File: src/main/java/duke/FindMessage.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+                searchResults.add(task);
+            }
+        }
+        int count = 1;
+        System.out.println("Here are the matching tasks in your list:\n");
+        for(Task task : searchResults) {
+            System.out.println(count + "." + task.getTaskString());
+            count = count + 1;
+        }
+    }
```
Comment (eeliana) [inline src/main/java/duke/FindMessage.java]: Maybe you can add a blank line of separation between the logical units within this block and possibly other places too?

File: src/main/java/duke/TaskFactory.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+public class TaskFactory {
+    /**
+     * This is the method responsible for instantiating the right subtype of task.
+     * @param taskStr The string input from the user to be stored as a task
+     * @return Task A task object with the user's
+     * @exception IllegalFormatException Incorrect format for creating a task, prompts to correct format
+     * @exception EmptyDescriptionException If the description of a todo task is empty
+     * @exception InvalidCommandException If the taskStr doesn't match any task's format at all
+     */
+    public static Task createTask(String taskStr) throws IllegalFormatException, EmptyDescriptionException, InvalidCommandException {
```
Comment (eeliana) [inline src/main/java/duke/TaskFactory.java]: Perhaps a line break here could improve readability? Does this line have more than 120 chars?

**Review by eeliana — COMMENTED** [review]
Overall, some public methods and classes are missing documentation and maybe some areas where line breaks could improve readability. The code was generally not hard to read and understand but there are a few coding standard violations to fix. LGTM, after fixing these violations.

File: src/main/java/duke/DeleteTaskMessage.java (hunk 2 above)
Comment (SajalVaishnav) [reply src/main/java/duke/DeleteTaskMessage.java]: Haha well yes... but prof said we can choose a personality and I chose an angry indian uncle... now there's no turning back

File: src/main/java/duke/Deadline.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
+package duke;
+
+import java.time.format.DateTimeFormatter;
+import java.time.LocalDate;
+
+/**
+ * The class for creating and managing Deadlines added to TaskList. Extends Task.
+ */
+public class Deadline extends Task {
+    private boolean done;
```
Comment (bnjmnt4n) [inline src/main/java/duke/Deadline.java]: I think `done` can be renamed to `isDone` to better indicate that it is a boolean.

File: src/main/java/duke/Duke.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
+package duke;
+
+import java.io.IOException;
+
+/**
+ * The main class running Duke.
+ */
+public class Duke {
+    //true indicates the conversation is ongoing
+    public static boolean conversationState;
```
Comment (bnjmnt4n) [inline src/main/java/duke/Duke.java]: Do you think `conversationState` should be a publicly accessible static field which all the messages can access? I think that breaks the abstraction of the `Duke` class.

File: src/main/java/duke/TaskFactory.java [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+        if(taskStr.substring(0, 4).equals("todo")) {
+            //create a todo task
+            try {
+                String taskDescription = taskStr.substring(5);
+                if(taskDescription.stripLeading().equals(""))
```
Comment (bnjmnt4n) [inline src/main/java/duke/TaskFactory.java]: Should this be using Egyptian braces instead?

**Review by bnjmnt4n — COMMENTED** [review]
The code is very readable, although there are some coding standards that are not met such as the Egyptian braces and variable naming. Also, I found the usage of `conversationState` a little confusing.

**Pull request closed without merging**